core/src/main/java/io/fabric8/maven/core/model/GroupArtifactVersion.java

```diff
@@ -42,4 +44,18 @@ public String getVersion() {
     public boolean isSnapshot() {
         return getVersion() != null && getVersion().endsWith("SNAPSHOT");
     }
+
+    /**
+     * ArtifactId is used for setting a resource name (service, pod,...) in Kubernetes resource.
+     * The problem is that a Kubernetes resource name must start by a char.
+     * This method returns a valid string to be used as Kubernetes name.
+     * @return Sanititzed Kubernetes name.
+     */
+    public String getSanitizedArtifactId() {
```

Comment:
I think apart from checking whether artifactId starts with a digit we should also sanitize it based on regex: `[a-z0-9]([-a-z0-9]*[a-z0-9])?` and max length(i.e 63 characters). WDYT?

Reply:
I thought the same but, look if image name starts with a number there is an easy fix (put a char), but in case of more than 63 chars (we cannot do anything, better to not substring a full name because probably is to avoid any duplicated). And again any other char, what would be the solution?

For this reason, I think that the best is to fix what we can and leave as an exception the other things.

But of course, it is an opinion.

Reply:
ah, yeah. Makes sense... as always ;-)

Reply:
Well most of the time I say things that has no sense at all :)